src/sentry/tasks/post_process.py

```diff
@@ -703,15 +703,11 @@ def process_inbox_adds(job: PostProcessJob) -> None:
                         event.group.substatus = GroupSubStatus.NEW
                         add_group_to_inbox(event.group, GroupInboxReason.NEW)
                 elif is_regression:
-                    updated = (
-                        Group.objects.filter(id=event.group.id)
-                        .exclude(substatus=GroupSubStatus.REGRESSED)
-                        .update(status=GroupStatus.UNRESOLVED, substatus=GroupSubStatus.REGRESSED)
-                    )
```

Comment:
This was always returning false since the group would already be updated to the correct regressed state here https://github.com/getsentry/sentry/blob/f5a40a2530a1167dc60b4158c81a611521771bdd/src/sentry/event_manager.py#L1755-L1775

Since the row was already updated, the conditional update would not update anything leading to `add_group_to_inbox` never being called.